scripts/check-go-vulnerabilities.sh

```diff
@@ -24,14 +33,117 @@ fi
+all_vulnerabilities=$(echo "$result" | jq -s '
+    [.[] | select(.finding) | {
+        osv: .finding.osv,
+        fixed_version: (.finding.fixed_version // "N/A"),
+        module: (.finding.trace[0].module // "unknown")
+    }] | unique_by(.osv)
+' 2>/dev/null)
+
+# Build case-insensitive regex pattern from ignored packages
+if [ ${#IGNORED_PACKAGES[@]} -gt 0 ]; then
+    ignored_pattern=$(printf "%s\n" "${IGNORED_PACKAGES[@]}" | awk '{print tolower($0)}' | paste -sd '|' -)
+else
+    ignored_pattern=""
+fi
+
+# Split vulnerabilities: fixes available, no fixes available, and ignored
+if [ -n "$ignored_pattern" ]; then
+    fixes_available=$(echo "$all_vulnerabilities" | jq --arg ignored "$ignored_pattern" '
+        [.[] | select(.fixed_version != "N/A") |
+         select((.module | ascii_downcase | test($ignored)) | not)]
+    ')
+    no_fixes_available=$(echo "$all_vulnerabilities" | jq --arg ignored "$ignored_pattern" '
+        [.[] | select(.fixed_version == "N/A") |
+         select((.module | ascii_downcase | test($ignored)) | not)]
+    ')
+    ignored_vulnerabilities=$(echo "$all_vulnerabilities" | jq --arg ignored "$ignored_pattern" '
+        [.[] | select(.module | ascii_downcase | test($ignored))]
+    ')
 else
-    echo "Please run govulncheck to check for vulnerabilities. See below for found vulnerabilities."
-    version=$($govul --version)
-    echo "Currently using govulncheck version: $version"
-    echo "$result"
+    fixes_available=$(echo "$all_vulnerabilities" | jq '[.[] | select(.fixed_version != "N/A")]')
+    no_fixes_available=$(echo "$all_vulnerabilities" | jq '[.[] | select(.fixed_version == "N/A")]')
+    ignored_vulnerabilities='[]'
+fi
+
+# Count vulnerabilities in each category
+fixes_available_count=$(echo "$fixes_available" | jq 'length')
+no_fixes_available_count=$(echo "$no_fixes_available" | jq 'length')
+ignored_vulnerabilities_count=$(echo "$ignored_vulnerabilities" | jq 'length')
+
+# Display results
+echo ""
+echo "=========================================="
+echo "VULNERABILITY SCAN RESULTS"
+echo "=========================================="
+echo ""
+
+# Section 1: Vulnerabilities with fixes available (FAIL if found)
+echo "1. VULNERABILITIES WITH FIXES AVAILABLE ($fixes_available_count)"
+[ "$fixes_available_count" -gt 0 ] && echo "   Status: FAIL - MUST FIX BELOW VULNERABILITIES" || echo "   Status: PASS"
+if [ "$fixes_available_count" -gt 0 ]; then
+    echo "$fixes_available" | jq -r 'group_by(.module) | .[] |
+        "  Module: \(.[0].module)\n" +
+        (map("    • \(.osv) (Fix: \(.fixed_version))") | join("\n")) + "\n"'
+else
+    echo "  None"
+fi
+echo "-------------------------------------------"
+
+# Section 2: Vulnerabilities with no fixes currently available (PASS)
+echo "2. VULNERABILITIES WITH NO FIXES CURRENTLY AVAILABLE ($no_fixes_available_count)"
+echo "   Status: PASS - No fix released yet, create tickets to track"
```

Comment:
I imagine if there's no fix released, we'll ignore these results because the task will succeed. Is the plan to ignore vulnerabilities that have no fix, or somehow remove them?